integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/util/SecondaryIndexUtil.scala

```diff
@@ -280,6 +286,23 @@ object SecondaryIndexUtil {
     }
   }
 
+  /**
+   * This method deletes the old index files or merge index file after data files merge
+   */
+  private def deleteOldIndexOrMergeIndexFiles(
+      factTimeStamp: Long,
+      validSegments: util.List[Segment],
+      indexCarbonTable: CarbonTable): Unit = {
+    // delete the index/merge index carbonFile of old data files
+    validSegments.asScala.foreach { segment =>
+      indexCarbonTable.getIndexFilesListForSegment(segment).asScala.filter { indexFile =>
+        DataFileUtil.getTimeStampFromFileName(indexFile).toLong > factTimeStamp
+      }.foreach { indexFile =>
```

Comment:
combine filter and foreach into a single foreach

Reply:
done